### PR TITLE
UUID specs

### PR DESCRIPTION
**Pull request by Corniel**
Rewrote UUID tests into specs.
Added guard on min and max date for generation sequential UUID's.

### CODE REVIEW

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+        {
+            TypeConverterAssert.ConvertToEquals(Svo.Guid, Svo.Uuid);
+        }
+    }
+
+    public class Supports_JSON_serialization
+    {
+        [TestCase(null, "")]
+        [TestCase("Qowaiv_SVOLibrary_GUIA", "Qowaiv_SVOLibrary_GUIA")]
+        public void convension_based_deserialization(Uuid expected, object json)
```
Comment (LauraXiulan) [inline test/Qowaiv.UnitTests/UUID_specs.cs:612]: convention?

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+        {
+            var actual = JsonTester.Read<Uuid>(json);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [TestCase(null, "")]
+        [TestCase("Qowaiv_SVOLibrary_GUIA", "Qowaiv_SVOLibrary_GUIA")]
+        public void convension_based_serialization(object expected, Uuid svo)
```
Comment (LauraXiulan) [inline test/Qowaiv.UnitTests/UUID_specs.cs:620]: See comment above

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+        }
+
+        [Test]
+        public void to_Uuid_only()
+        {
+            Assert.Throws<ArgumentException>(() => Svo.Uuid.CompareTo(new object()));
+        }
+
+        [Test]
+        public void can_be_sorted()
```
[the remark this answers is not in the document]
Comment (Corniel) [reply test/Qowaiv.UnitTests/UUID_specs.cs:503]: I thought so, do you have a suggestion?

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 4]
```diff
@@ -0,0 +1,707 @@
...
+
+        [Test]
+        public void export_to_byte_array_equal_to_GUID_equivalent()
+        {
+            var bytes = Svo.Uuid.ToByteArray();
+            Assert.AreEqual(((Guid)Svo.Uuid).ToByteArray(), bytes);
+        }
+    }
+
+    public class Debug_experience
```
[the remark this answers is not in the document]
Comment (Corniel) [reply test/Qowaiv.UnitTests/UUID_specs.cs:697]: It tries to explain that there is support for displaying the SVO content/value in the debugger. Alternative naming for this is welcome.

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 5]
```diff
@@ -0,0 +1,707 @@
...
+        public void with_TryParse_returns_SVO()
+        {
+            Assert.AreEqual(Svo.Uuid, Uuid.TryParse("Qowaiv_SVOLibrary_GUIA"));
+        }
+    }
+
+    public class Can_be_created
+    {
+        [Test]
+        public void Randomly()
```
[the remark this answers is not in the document]
Comment (Corniel) [reply test/Qowaiv.UnitTests/UUID_specs.cs:220]: that a UUID can be created with a random (likely unique) value.

File: test/Qowaiv.UnitTests/UUID_specs.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (LauraXiulan) [reply test/Qowaiv.UnitTests/UUID_specs.cs:503]: I think I would expect a separate class like Can_be_parsed, but seeing as there is only one test for it, I am not sure.

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 6]
```diff
@@ -0,0 +1,707 @@
...
+
+        [Test]
+        public void with_SHA1()
+        {
+            var hashed = Uuid.GenerateWithSHA1(Encoding.UTF8.GetBytes("Qowaiv"));
+            Assert.AreEqual(Uuid.Parse("39h-Y1rR51ym_t78x9h0bA"), hashed);
+        }
+    }
+
+    public class Can_be_created_squential
```
[the remark this answers is not in the document]
Comment (Corniel) [reply test/Qowaiv.UnitTests/UUID_specs.cs:240]: Well, it tests that 1 tick before that boundary fails, to proof that the boundary works. Writing two tests for it felt a bit overdone.

File: test/Qowaiv.UnitTests/UUID_specs.cs [hunk 7]
```diff
@@ -0,0 +1,707 @@
...
+        [Test]
+        public void until_3_Dec_9276()
+        {
+            using (Clock.SetTimeForCurrentThread(() => new DateTime(9276, 12, 04)))
+            {
+                Assert.Catch<InvalidOperationException>(() => Uuid.NewSequential());
+            }
+        }
+
+        /// <remarks>Due to the reordening the version ands up in index 7.</remarks>
```
[the remark this answers is not in the document]
Comment (Corniel) [reply test/Qowaiv.UnitTests/UUID_specs.cs]: I removed it. It is not part of the story, so I dropped it.

**Pull request merged**